index.js

```diff
@@ -234,7 +237,7 @@ export default class TextMarquee extends PureComponent {
 
   invalidateMetrics = () => {
     this.distance = null
-    this.setState({ contentFits: false })
+    this.setState({ contentFits: true })
```

Comment:
Can you explain the data flow behind this change? (line 70 too). Seems like a pretty big fundamental change to flip the defaults for contentFits.

Reply:
Oh yeah sorry I didn't really comment on this change, it's more a fix of something that I observed, a bug that appeared with the implementation of this new prop `shouldAnimateTreshold`:

**The bug:** For a text that fits inside its container, the text would appear twice (I mean spatially, like that: "text    text") initially because of this initial state with `state.contentFits` value to false, and only after the layout callback of the scrollview (`onContentSizeChange`) would the proper value of `contentFits` be computed. So for like the first frames, there would be this duplicated text, due to line 305.

**Why this bug appeared:** I think this wasn't an issue before, because for a "text that fits", the container would be the size of the text, so the "duplicate" would be out of the container, thus not drawn to the screen.
With this new prop `shouldAnimateTreshold`, I had to put `flex: 1` to the container view to make it take all the space available to it because otherwise it was impossible to compute the metrics properly (that's what I tried to explain line 269). That is this `flex: 1` that caused the bug I'm talking about (duplicated text), hence making it necessary to change the initial value of `state.contentFits`.

**Is it a breaking though ?**
Changing the initial value of `state.contentFits` lets `calculateMetrics` decide for its value, only once the ScrollView is drawn. And by looking at the code, I don't think that assuming that `contentFits` is true at first would cause any error in calculating the metrics. But maybe there's a corner case I'm not seeing. At least in the example App everything looks as it should.

Reply:
Brilliant, I'll pull down the code later today and if all goes well I'll merge and push out a release.

Reply:
Great, thanks for your review!